errors.go

```diff
@@ -32,6 +32,11 @@ var (
 
 	// ErrZeroDimensional signifies an optimization was called with an input of length 0.
 	ErrZeroDimensional = errors.New("optimize: zero dimensional input")
+
+	// ErrNoProgress signifies that Linesearch cannot make further progress
+	// because the next evaluation point is equal to the current one due to
+	// floating-point arithmetic.
+	ErrNoProgress = errors.New("linesearch: no progress")
```

Comment:
Could we make this more informative? "linesearch: no change in location after linesearch step"